webapp/src/components/systemConsole/surveyResults/surveyResults.tsx

```diff
@@ -35,7 +38,39 @@ function SurveyResults() {
         fetchSurveyStatus();
     }, [hydrateSurveyResults]);
 
-    const generateActions = (surveyResult: SurveyResult) => {
+    const handleStopSurvey = useCallback(async () => {
+        if (!currentSurveyID.current) {
+            return;
+        }
+
+        const updatedSurveyResult = [...surveyResults];
```

Comment:
Nitpick: You don't need to create this list until you are going to set the state, so you can defer it until line 58. That way, if anything fails, you haven't created a new list "for nothing", which means less stress for the GC.

But yeah... probably not our bottleneck by far :stuck_out_tongue:

Reply:
Done! The GC thanks you for caring about it 🤣